src/field_map_change.c

```diff
@@ -241,57 +241,53 @@ void sub_020531A0 (FieldSystem * fieldSystem)
     gCoreSys.unk_65 = fieldSystem->unk_74->unk_00_12;
 }
 
-void sub_020531C0 (FieldSystem * fieldSystem, BOOL param1)
+void FieldSystem_MapChange_UpdateGameData (FieldSystem * fieldSystem, BOOL noWarp)
 {
-    int v0 = fieldSystem->unk_1C->unk_00;
-    UnkStruct_0203A790 * v1 = sub_0203A790(fieldSystem->saveData);
+    int mapId = fieldSystem->location->mapId;
+    FieldStatus * fieldStatus = SaveData_GetFieldStatus(fieldSystem->saveData);
 
-    sub_02055414(fieldSystem);
-    sub_0203F1C4(fieldSystem);
+    Sound_ClearSpecialBGM(fieldSystem);
+    FieldSystem_ClearLocalFlags(fieldSystem);
 
-    if (!param1) {
-        sub_02070494(fieldSystem);
+    if (!noWarp) { 
+        FieldSystem_InitFlagsWarp(fieldSystem);
     } else {
-        sub_02070430(fieldSystem);
+        FieldSystem_InitFlagsNoWarp(fieldSystem);
     }
 
-    sub_0206B0D4(SaveData_Events(fieldSystem->saveData));
+    FieldEvents_ResetVSSeeker(SaveData_GetFieldEvents(fieldSystem->saveData));
 
-    if (!param1) {
+    if (!noWarp) {
         sub_020559DC(fieldSystem);
     }
 
-    if (!param1) {
+    if (!noWarp) {
         sub_02027F50(sub_02027860(fieldSystem->saveData));
     }
+    
+    FieldEvents * events = SaveData_GetFieldEvents(fieldSystem->saveData);
+    u16 weather = FieldSystem_GetWeather(fieldSystem, mapId);
 
-    {
-        UnkStruct_020507E4 * v2 = SaveData_Events(fieldSystem->saveData);
-        u16 v3 = sub_0203A944(fieldSystem, v0);
-
-        if (((v3 == 14) && (sub_0206AF0C(v2) == 1)) || ((v3 == 16) && (sub_0206AEDC(v2) == 1))) {
-            v3 = 0;
-        }
-
-        sub_0203A754(v1, v3);
+    if (((weather == 14) && (FieldEvents_CheckDefog(events) == 1)) || ((weather == 16) && (FieldEvents_CheckFlash(events) == 1))) {
+        weather = 0;
     }
```

Comment:
We have constants for these values already, but they're in the battle overlay headers: https://github.com/pret/pokeplatinum/blob/f747c2fa6e5fe15b50924916d09af58c46d035ba/include/constants/battle/condition.h#L149

Could you move them to somewhere accessible by both battle and field code as part of this PR?